lnrpc/invoicesrpc/addinvoice.go

```diff
@@ -253,6 +258,16 @@ func AddInvoice(ctx context.Context, cfg *AddInvoiceConfig,
 		options = append(options, zpay32.CLTVExpiry(uint64(defaultDelta)))
 	}
 
+	if len(invoice.PrivateChannels) > 0 && !invoice.Private {
```

Comment:
Wouldn't the presence of a list of private channels imply that the user wants to have a private invoice? 
Non-critical, but I can definitely see myself calling with a list of channels and forgetting the private flag and wanting it to be set anyway.